stdlib/std.jsonnet

```diff
@@ -1041,7 +1042,43 @@ limitations under the License.
         std.join('', lines);
     aux(value, [], ''),
 
-  manifestYamlDoc(value, indent_array_in_object=false)::
+  manifestYamlDoc(value, indent_array_in_object=false, quote_keys=true)::
+    local isReserved(key) =
+      local reserved = ['true', 'false', 'null'];
+      local bad = [word for word in reserved if word == key];
+      if std.length(bad) > 0 then
+        true
+      else false;
+    local bareSafe(key) =
+      # Letters has '_' and '-' because they can be treated equally
+      local letters = std.set(std.stringChars('ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz_-'));
```

Comment:
`---` (document separator) and `-` (sequence item indicator) must appear on its own line or be followed by a whitespace, so it should be safe to use as a mapping key as in `---: 1`. But it might be better for us to check the major parser implemantations.

Reply:
For `-`, also note that parsers may recognize timestamp values (e.g. `2021-05-25`): http://yaml.org/type/timestamp.html

Reply:
`-` is now quoted as documented in the spec.

As hanazuki states, the yaml spec does not state that `---` is an invalid identifier, and at least the python yaml implementation parses `---`correctly on input
```bash
mwilder@Matts-MBP ~> python3 -c 'import yaml; print(yaml.safe_load("---: 3dashes\n--: 2dashes\nfoo: bar\n"))'
{'---': '3dashes', '--': '2dashes', 'foo': 'bar'}
```
(note: `--` included for reference)

However it does quote it on output
```bash
mwilder@Matts-MBP ~> python3 -c 'import yaml; print(yaml.dump({"---": "3dashes", "--": "2dashes", "foo": "bar"}))'
--: 2dashes
'---': 3dashes
foo: bar

```

Go exhibits the same behavior: https://play.golang.org/p/0RkOEarVOas

Since both python and go both quote `---` identifiers on output, I am inclined to do the same thing, despite it being not required by the spec.

Reply:
> For `-`, also note that parsers may recognize timestamp values (e.g. `2021-05-25`): http://yaml.org/type/timestamp.html

Dates are now handled correctly, with comprehensive tests in place to demonstrate this.